src/AutoRest.CSharp/MgmtTest/Generation/MgmtBaseTestWriter.cs

```diff
@@ -769,5 +769,36 @@ public CSharpType GenResponseType(MgmtClientOperation clientOperation, bool asyn
             }
             return null;
         }
+
+        protected void WriteMethodInvocation(string methodName, List<string> paramNames)
+        {
+            _writer.Append($"{methodName}(");
+            foreach (var paramName in paramNames)
+            {
+                _writer.Append($"{paramName},");
+            }
+            _writer.RemoveTrailingComma();
+            _writer.Append($")");
+        }
+
+        protected void WriteMethodTestInvocation(bool async, bool isPagingOperation, string methodName, List<string> paramNames)
+        {
+            _writer.Append($"{GetAwait(async)}");
+            if (isPagingOperation)
+            {
+                _writer.Append($"foreach (var _ in ");
+            }
+            WriteMethodInvocation($"{methodName}", paramNames);
+            if (isPagingOperation)
+            {
+                _writer.Append($")");
+                using (_writer.Scope())
+                { }
+            }
+            else
+            {
+                _writer.Line($";");
+            }
```

Comment:
I suggest we could completely separate `isPagingOperation` and not paging operation branches, like this:
```suggestion
            if (isPagingOperation)
            {
            // write the paging case
            }
            else
            {
            // write non-paging case
            }
```